python/sglang/srt/layers/layernorm.py

```diff
@@ -52,12 +52,8 @@ def __init__(
     def forward(self, *args, **kwargs):
         if torch.compiler.is_compiling():
             return self.forward_native(*args, **kwargs)
-        if _is_cuda:
-            return self.forward_cuda(*args, **kwargs)
-        elif _is_hip:
-            return self.forward_hip(*args, **kwargs)
         else:
-            return self.forward_native(*args, **kwargs)
+            return self._forward_method(*args, **kwargs)
```

Comment:
nit: wondering whether it is possible we just completely remove all the forward methods?

btw, this refactor may be related: https://github.com/sgl-project/sglang/pull/5382

Reply:
> nit: wondering whether it is possible we just completely remove all the forward methods?
> 
> btw, this refactor may be related: #5382

Looks good! We can record the origin _forwrd_method in enter_torch_compile and recover it in leave_torch_compile, rather than recovering to forward_cuda

Reply:
Pretty reasonable

Reply:
what about firstly merging 5382, and then do a modifcation

I personally like to have a refactor PR + a change PR, o/w a PR with both refactor and change will make it pretty hard to trace